persistence/module.go

```diff
@@ -36,13 +37,14 @@ type persistenceModule struct {
 	// A key-value store mapping heights to blocks. Needed for block synchronization.
 	blockStore blockstore.BlockStore
 
-	// A tx indexer (i.e. key-value store) mapping transaction hashes to transactions. Needed for
-	// avoiding tx replays attacks, and is also used as the backing database for the transaction
-	// tx merkle tree.
+	// txIndexer is a key-value store mapping transaction hashes to transactions.
+	// It is needed for avoiding tx replays attacks, and is also used as the backing
+	// data store for tree store.
```

Comment:
```suggestion
	// It is needed for avoiding tx replays attacks, and some business logic around tx validation.
	// IMPORTANT: It doubles as the backing data store for the tx tree in one of the `stateTrees`.
```

Reply:
NB: The connection to the `TxIndexer` is covered in the design document for this refactor.